src/ColorGrid.tsx

```diff
@@ -0,0 +1,33 @@
+import type React from 'react';
+import styles from './ColorGrid.module.css';
+import colorData from './colors.json';
+
+interface ComplementaryColor {
+    hex: string;
+    name: string;
+}
+
+interface Color {
+    hex: string;
+    name: string;
+    comp: ComplementaryColor[];
```

Comment:
you could define Color with an optional array of Color instead of defining another type for ComplementaryColor